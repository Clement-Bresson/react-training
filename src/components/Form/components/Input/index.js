import React, { Component } from 'react'
import { PropTypes as T } from 'prop-types'

class Input extends Component {

  state = {
    value: ''
  }

  handleChange = event => {
    const { onChange, name } = this.props
    const value = event.target.value
    this.setState({ value })
    onChange && onChange(name, value)
  }

  render() {
    const { name, type } = this.props
    const { value } = this.state
    return (
      <input
        name={name}
        onChange={this.handleChange}
        type={type}
        value={value}
      />
    )
  }
}

Input.defaultProps = {
  type: 'text'
}

Input.propTypes = {
  name: T.string.isRequired,
  onChange: T.func.isRequired,
  type: T.string,
  value: T.string
}

export default Input
