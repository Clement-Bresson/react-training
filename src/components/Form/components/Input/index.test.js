import Input from './'
import React from 'react'
import { shallow } from 'enzyme'

let component
let onChange

describe('Input', () => {

  beforeEach(() => {
    onChange = jest.fn()
    component = shallow(<Input name='bar' onChange={onChange} />)
  })

  afterEach(() => {
    component.unmount()
  })

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('onChange', () => {
    const input = component.find('input')
    expect(input).toBeDefined()
    input.simulate('change', { target: { value: 'foo' } })
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toHaveBeenCalledWith('bar', 'foo')
  })
})
