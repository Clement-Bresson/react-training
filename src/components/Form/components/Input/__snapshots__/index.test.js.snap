// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    name="bar"
    onChange={[MockFunction]}
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "name": "bar",
      "onChange": [Function],
      "type": "text",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "name": "bar",
        "onChange": [Function],
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
